src/internal/list.js

```diff
@@ -0,0 +1,22 @@
+function List(head, tail){
+  this.head = head;
+  this.tail = tail;
+  this.size = tail.size + 1;
+}
```

Comment:
@safareli, you once suggested using a List over an Array. You may be interesting in seeing it. :)